commit a3fe275e762372c19d01cf8286a5952f01f2dc2e
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Thu Dec 13 17:28:24 2018 +0100

    Remove spec file and srpm targets

diff --git a/meson.build b/meson.build
index db7c6a1..2a46f39 100644
--- a/meson.build
+++ b/meson.build
@@ -51,51 +51,4 @@ gobject = dependency('gobject-2.0')
 yaml = dependency('yaml-0.1')
 gtkdoc = dependency('gtk-doc')
 
-spec_tmpl = find_program('spec_tmpl.sh')
-
-specfile_template = files('libmodulemd.spec.in')
-
-mkdir = find_program('mkdir')
-
-rpmsetup_target = custom_target(
-    'rpmsetup',
-    command: [
-        mkdir, '-p',
-        'rpmbuild/BUILD',
-        'rpmbuild/RPMS',
-        'rpmbuild/SPECS',
-        'rpmbuild/SRPMS'],
-    output: 'rpmbuild',
-)
-
-spec_target = custom_target(
-    'specfile',
-    capture: true,
-    build_always: true,
-    command: [spec_tmpl, meson.project_version(), '@INPUT@'],
-    input: specfile_template,
-    output: 'libmodulemd.spec',
-    depends: rpmsetup_target,
-)
-
-rpm_cdata = configuration_data()
-rpm_cdata.set('VERSION', meson.project_version())
-rpm_cdata.set('BUILDFLAG', '-bb')
-
-srpm_cdata = configuration_data()
-srpm_cdata.set('VERSION', meson.project_version())
-srpm_cdata.set('BUILDFLAG', '-bs')
-
-configure_file(
-    input: 'make_rpms.sh.in',
-    output: 'make_srpm.sh',
-    configuration: srpm_cdata,
-)
-
-configure_file(
-    input: 'make_rpms.sh.in',
-    output: 'make_rpms.sh',
-    configuration: rpm_cdata,
-)
-
 subdir('modulemd')
