commit 7f67d3690caf9a87fa4486d0da786e4f5dbb92c0
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Thu Sep 14 14:19:20 2017 +0200

    Make data import more robust

diff --git a/modules/attack/wplfi.py b/modules/attack/wplfi.py
index 67b7676..5f39e61 100644
--- a/modules/attack/wplfi.py
+++ b/modules/attack/wplfi.py
@@ -19,8 +19,10 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp 
+from lib import wphttp
 from lib import wpprint
+import inspect
+import os.path
 import re
 import urllib 
 
@@ -38,7 +40,7 @@ class wplfi:
 		print ""
 		params = dict([x.split("=") for x in self.payload.split("&")])
 		param = {}
-		db = open("data/wplfi.txt","rb")
+		db = open(os.path.join(os.path.dirname(inspect.getsourcefile(self.__class__)),"../../data/wplfi.txt"),"rb")
 		file = [x.split("\n") for x in db]
 		try:
 			for item in params.items():
@@ -53,4 +55,4 @@ class wplfi:
 						self.printf.plus("[%s][%s][not vuln] %s"%(resp.getcode(),self.method,resp.geturl()))
 					param[item[0]] = item[1].replace(x[0],item[1])
 		except Exception as error:
-			pass
\ No newline at end of file
+			pass
diff --git a/modules/attack/wpsql.py b/modules/attack/wpsql.py
index bd32b49..121bc15 100644
--- a/modules/attack/wpsql.py
+++ b/modules/attack/wpsql.py
@@ -19,8 +19,10 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp 
+from lib import wphttp
 from lib import wpprint
+import inspect
+import os.path
 import re
 import urllib 
 
@@ -68,7 +70,7 @@ class wpsql:
 		print ""
 		params = dict([x.split("=") for x in self.payload.split("&")])
 		param = {}
-		db = open("data/wpsql.txt","rb")
+		db = open(os.path.join(os.path.dirname(inspect.getsourcefile(self.__class__)),"../../data/wpsql.txt"),"rb")
 		file = [x.split("\n") for x in db]
 		try:
 			for item in params.items():
@@ -84,4 +86,4 @@ class wpsql:
 						self.printf.plus("[%s][%s][not vuln] %s"%(resp.getcode(),self.method,resp.geturl()))
 					param[item[0]] = item[1].replace(x[0],item[1])
 		except Exception as error:
-			pass
\ No newline at end of file
+			pass
diff --git a/modules/attack/wpxss.py b/modules/attack/wpxss.py
index 12f6fd9..aa58cb2 100644
--- a/modules/attack/wpxss.py
+++ b/modules/attack/wpxss.py
@@ -19,11 +19,14 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp 
+from lib import wphttp
 from lib import wpprint
+import inspect
+import os.path
 import re
 import urllib 
 
+
 class wpxss:
 	check = wphttp.check()
 	printf = wpprint.wpprint()
@@ -38,7 +41,7 @@ class wpxss:
 		print ""
 		params = dict([x.split("=") for x in self.payload.split("&")])
 		param = {}
-		db = open("data/wpxss.txt","rb")
+		db = open(os.path.join(os.path.dirname(inspect.getsourcefile(self.__class__)),"../../data/wpxss.txt"),"rb")
 		file = [x.split("\n") for x in db]
 		try:
 			for item in params.items():
