commit 19623c1f4d4332649ac32de88ed9e4e9ff5a743a
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Thu Sep 14 14:23:37 2017 +0200

    Add setup.py packaging

diff --git a/wpseku.py b/bin/wpseku
similarity index 95%
rename from wpseku.py
rename to bin/wpseku
index f6e95d6..898795e 100644
--- a/wpseku.py
+++ b/bin/wpseku
@@ -21,19 +21,19 @@
 # Add path to seft.target
 
 
-from lib import wpcolor
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wpcolor
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import os
 import sys
 import getopt
 import time
 import urlparse
-from modules.discovery import wpall
-from modules.bruteforce import wpxmlrpc
-from modules.attack import wpxss
-from modules.attack import wpsql
-from modules.attack import wplfi
+from wpseku.modules.discovery import wpall
+from wpseku.modules.bruteforce import wpxmlrpc
+from wpseku.modules.attack import wpxss
+from wpseku.modules.attack import wpsql
+from wpseku.modules.attack import wplfi
 
 class WPSeku(object):
 	"""docstring for WPSeku"""
diff --git a/modules/attack/wplfi.py b/modules/attack/wplfi.py
index 5f39e61..8b20c64 100644
--- a/modules/attack/wplfi.py
+++ b/modules/attack/wplfi.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import inspect
 import os.path
 import re
diff --git a/modules/attack/wpsql.py b/modules/attack/wpsql.py
index 121bc15..f65efde 100644
--- a/modules/attack/wpsql.py
+++ b/modules/attack/wpsql.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import inspect
 import os.path
 import re
diff --git a/modules/attack/wpxss.py b/modules/attack/wpxss.py
index aa58cb2..48cd456 100644
--- a/modules/attack/wpxss.py
+++ b/modules/attack/wpxss.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import inspect
 import os.path
 import re
diff --git a/modules/bruteforce/wpxmlrpc.py b/modules/bruteforce/wpxmlrpc.py
index e8a7086..157a6ac 100644
--- a/modules/bruteforce/wpxmlrpc.py
+++ b/modules/bruteforce/wpxmlrpc.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp 
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import re 
 
 class wpxmlrpc:
@@ -53,4 +53,4 @@ class wpxmlrpc:
 				else:
 					self.printf.erro('Invalid Credentials: \"%s\" - \"%s\"'%(self.user,x.split('\n')[0]))
 			except Exception as error:
-				pass
\ No newline at end of file
+				pass
diff --git a/modules/discovery/generic/wpconfig.py b/modules/discovery/generic/wpconfig.py
index 47e12e3..2fed65e 100644
--- a/modules/discovery/generic/wpconfig.py
+++ b/modules/discovery/generic/wpconfig.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp 
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import re
 
 class wpconfig:
diff --git a/modules/discovery/generic/wpcrossdomain.py b/modules/discovery/generic/wpcrossdomain.py
index 0b1e8dc..3fbea6f 100644
--- a/modules/discovery/generic/wpcrossdomain.py
+++ b/modules/discovery/generic/wpcrossdomain.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 
 class wpcrossdomain:
 	check = wphttp.check()
@@ -39,4 +39,4 @@ class wpcrossdomain:
 			else:
 				self.printf.erro('crossdomain.xml not available')
 		except Exception as error:
-			pass
\ No newline at end of file
+			pass
diff --git a/modules/discovery/generic/wpfpd.py b/modules/discovery/generic/wpfpd.py
index 26469d8..8b1b5b6 100644
--- a/modules/discovery/generic/wpfpd.py
+++ b/modules/discovery/generic/wpfpd.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp 
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import re
 
 class wpfpd:
diff --git a/modules/discovery/generic/wpheaders.py b/modules/discovery/generic/wpheaders.py
index 6da7b9c..8475b67 100644
--- a/modules/discovery/generic/wpheaders.py
+++ b/modules/discovery/generic/wpheaders.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 
 class wpheaders:
     check = wphttp.check()
@@ -171,4 +171,4 @@ class wpheaders:
     			print 'X-XSS-Protection: %s'%(r.info().getheader('x-xss-protection'))
     	except Exception as error:
     		pass
-        print ""
\ No newline at end of file
+        print ""
diff --git a/modules/discovery/generic/wphtaccess.py b/modules/discovery/generic/wphtaccess.py
index 3a02643..4803c4e 100644
--- a/modules/discovery/generic/wphtaccess.py
+++ b/modules/discovery/generic/wphtaccess.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 
 class wphtaccess:
 	check = wphttp.check()
@@ -39,4 +39,4 @@ class wphtaccess:
 			else:
 				self.printf.erro('.htaccess not available')
 		except Exception as error:
-			pass
\ No newline at end of file
+			pass
diff --git a/modules/discovery/generic/wplicense.py b/modules/discovery/generic/wplicense.py
index 0094f08..e918576 100644
--- a/modules/discovery/generic/wplicense.py
+++ b/modules/discovery/generic/wplicense.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp 
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 
 class wplicense:
 	check = wphttp.check()
diff --git a/modules/discovery/generic/wplisting.py b/modules/discovery/generic/wplisting.py
index 5e0231b..837be84 100644
--- a/modules/discovery/generic/wplisting.py
+++ b/modules/discovery/generic/wplisting.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import re
 
 class wplisting:
diff --git a/modules/discovery/generic/wploginprotection.py b/modules/discovery/generic/wploginprotection.py
index b987b6b..642945a 100644
--- a/modules/discovery/generic/wploginprotection.py
+++ b/modules/discovery/generic/wploginprotection.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import re
 
 class wploginprotection:
@@ -40,4 +40,4 @@ class wploginprotection:
 			elif resp.getcode() == 404:
 				self.printf.erro('wp-login detect protection')
 		except Exception as error:
-			pass
\ No newline at end of file
+			pass
diff --git a/modules/discovery/generic/wpreadme.py b/modules/discovery/generic/wpreadme.py
index bba0c1c..5d98239 100644
--- a/modules/discovery/generic/wpreadme.py
+++ b/modules/discovery/generic/wpreadme.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 
 class wpreadme:
 	check = wphttp.check()
diff --git a/modules/discovery/generic/wprobots.py b/modules/discovery/generic/wprobots.py
index 20f1153..c6c21c2 100644
--- a/modules/discovery/generic/wprobots.py
+++ b/modules/discovery/generic/wprobots.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp 
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 
 class wprobots:
 	check = wphttp.check()
diff --git a/modules/discovery/generic/wpsitemap.py b/modules/discovery/generic/wpsitemap.py
index a951c78..0becad4 100644
--- a/modules/discovery/generic/wpsitemap.py
+++ b/modules/discovery/generic/wpsitemap.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp 
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 
 class wpsitemap:
 	check = wphttp.check()
diff --git a/modules/discovery/generic/wpversion.py b/modules/discovery/generic/wpversion.py
index 9471232..05eed74 100644
--- a/modules/discovery/generic/wpversion.py
+++ b/modules/discovery/generic/wpversion.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import re
 import json
 import requests
@@ -121,4 +121,4 @@ class wpvuln:
 				self.printf.ipri('Not found vulnerabilities',color="g")
 				print ""
 		except Exception as error:
-			pass
\ No newline at end of file
+			pass
diff --git a/modules/discovery/generic/wpwaf.py b/modules/discovery/generic/wpwaf.py
index 4f15175..0528d60 100644
--- a/modules/discovery/generic/wpwaf.py
+++ b/modules/discovery/generic/wpwaf.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import re
 
 class wpwaf:
@@ -57,4 +57,4 @@ class wpwaf:
 			else:
 				self.printf.erro('No Firewall Detected')
 		except Exception as error:
-			pass
\ No newline at end of file
+			pass
diff --git a/modules/discovery/generic/wpxmlrpc.py b/modules/discovery/generic/wpxmlrpc.py
index 92746e6..792aca7 100644
--- a/modules/discovery/generic/wpxmlrpc.py
+++ b/modules/discovery/generic/wpxmlrpc.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp 
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import re
 
 class wpxmlrpc:
diff --git a/modules/discovery/plugins/wpchangelog.py b/modules/discovery/plugins/wpchangelog.py
index d2b180d..0726557 100644
--- a/modules/discovery/plugins/wpchangelog.py
+++ b/modules/discovery/plugins/wpchangelog.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 
 class wpchangelog:
 	check = wphttp.check()
@@ -38,4 +38,4 @@ class wpchangelog:
 				if resp.read() and resp.getcode() == 200:
 					self.printf.ipri('Changelog: %s'%(url),color="g")
 			except Exception as error:
-				pass
\ No newline at end of file
+				pass
diff --git a/modules/discovery/plugins/wplicense.py b/modules/discovery/plugins/wplicense.py
index 9a952b3..53716a2 100644
--- a/modules/discovery/plugins/wplicense.py
+++ b/modules/discovery/plugins/wplicense.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 
 class wplicense:
 	check = wphttp.check()
@@ -38,4 +38,4 @@ class wplicense:
 				if resp.read() and resp.getcode() == 200:
 					self.printf.ipri('License: %s'%(url),color="g")
 			except Exception as error:
-				pass
\ No newline at end of file
+				pass
diff --git a/modules/discovery/plugins/wplisting.py b/modules/discovery/plugins/wplisting.py
index e13c27a..8c388b3 100644
--- a/modules/discovery/plugins/wplisting.py
+++ b/modules/discovery/plugins/wplisting.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 
 class wplisting:
 	check = wphttp.check() 
@@ -38,4 +38,4 @@ class wplisting:
 				if re.search('Index of',resp.read()):
 					self.printf.ipri('Listing: %s'%(url),color="g")
 			except Exception as error:
-				pass
\ No newline at end of file
+				pass
diff --git a/modules/discovery/plugins/wpplugin.py b/modules/discovery/plugins/wpplugin.py
index bf1d0e2..cc5267e 100644
--- a/modules/discovery/plugins/wpplugin.py
+++ b/modules/discovery/plugins/wpplugin.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import re
 import json
 import requests
@@ -100,4 +100,4 @@ class wpvuln:
 			else:
 				self.printf.ipri('Not found vulnerabilities',color="g")
 		except Exception as error:
-			self.printf.ipri('Not found vulnerabilities',color="g")
\ No newline at end of file
+			self.printf.ipri('Not found vulnerabilities',color="g")
diff --git a/modules/discovery/plugins/wpreadme.py b/modules/discovery/plugins/wpreadme.py
index 9bb152e..bcbf1cc 100644
--- a/modules/discovery/plugins/wpreadme.py
+++ b/modules/discovery/plugins/wpreadme.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 
 class wpreadme:
 	check = wphttp.check()
@@ -38,4 +38,4 @@ class wpreadme:
 				if resp.read() and resp.getcode() == 200:
 					self.printf.ipri('Readme: %s'%(url),color="g")
 			except Exception as error:
-				pass
\ No newline at end of file
+				pass
diff --git a/modules/discovery/themes/wpchangelog.py b/modules/discovery/themes/wpchangelog.py
index 12d3f75..a794621 100644
--- a/modules/discovery/themes/wpchangelog.py
+++ b/modules/discovery/themes/wpchangelog.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 
 class wpchangelog:
 	check = wphttp.check()
@@ -38,4 +38,4 @@ class wpchangelog:
 				if resp.read() and resp.getcode() == 200:
 					self.printf.ipri('Changelog: %s'%(url),color="g")
 			except Exception as error:
-				pass
\ No newline at end of file
+				pass
diff --git a/modules/discovery/themes/wpfpd.py b/modules/discovery/themes/wpfpd.py
index 409f11e..bf784e2 100644
--- a/modules/discovery/themes/wpfpd.py
+++ b/modules/discovery/themes/wpfpd.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import re
 
 class wpfpd:
@@ -40,4 +40,4 @@ class wpfpd:
 				if re.search('Fatal error',resp.read()):
 					self.printf.ipri('Full Path Disclosure: %s'%(url),color="r")
 			except Exception as error:
-				pass
\ No newline at end of file
+				pass
diff --git a/modules/discovery/themes/wplicense.py b/modules/discovery/themes/wplicense.py
index e0e7fd4..22d27e7 100644
--- a/modules/discovery/themes/wplicense.py
+++ b/modules/discovery/themes/wplicense.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 
 class wplicense:
 	check = wphttp.check()
@@ -38,4 +38,4 @@ class wplicense:
 				if resp.read() and resp.getcode() == 200:
 					self.printf.ipri('License: %s'%(url),color="g")
 			except Exception as error:
-				pass
\ No newline at end of file
+				pass
diff --git a/modules/discovery/themes/wplisting.py b/modules/discovery/themes/wplisting.py
index f795bdf..815fbed 100644
--- a/modules/discovery/themes/wplisting.py
+++ b/modules/discovery/themes/wplisting.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import re
 
 class wplisting:
@@ -39,4 +39,4 @@ class wplisting:
 				if re.search('Index of',resp.read()):
 					self.printf.ipri('Listing: %s'%(url),color="r")
 			except Exception as error:
-				pass
\ No newline at end of file
+				pass
diff --git a/modules/discovery/themes/wpreadme.py b/modules/discovery/themes/wpreadme.py
index 9bb152e..bcbf1cc 100644
--- a/modules/discovery/themes/wpreadme.py
+++ b/modules/discovery/themes/wpreadme.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 
 class wpreadme:
 	check = wphttp.check()
@@ -38,4 +38,4 @@ class wpreadme:
 				if resp.read() and resp.getcode() == 200:
 					self.printf.ipri('Readme: %s'%(url),color="g")
 			except Exception as error:
-				pass
\ No newline at end of file
+				pass
diff --git a/modules/discovery/themes/wpstyle.py b/modules/discovery/themes/wpstyle.py
index 2bc47b0..34b89d5 100644
--- a/modules/discovery/themes/wpstyle.py
+++ b/modules/discovery/themes/wpstyle.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 
 class wpstyle:
 	check = wphttp.check()
@@ -36,4 +36,4 @@ class wpstyle:
 			if resp.read() and resp.getcode() == 200:
 				self.printf.ipri('Style: %s'%(url),color="g")
 		except Exception as error:
-			pass
\ No newline at end of file
+			pass
diff --git a/modules/discovery/themes/wptheme.py b/modules/discovery/themes/wptheme.py
index e3e7f32..aaaabe7 100644
--- a/modules/discovery/themes/wptheme.py
+++ b/modules/discovery/themes/wptheme.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import re
 import json
 import requests
@@ -106,4 +106,4 @@ class wpvuln:
 			else:
 				self.printf.ipri('Not found vulnerabilities',color="g")
 		except Exception as error:
-			self.printf.ipri('Not found vulnerabilities',color="g")
\ No newline at end of file
+			self.printf.ipri('Not found vulnerabilities',color="g")
diff --git a/modules/discovery/users/wpusers.py b/modules/discovery/users/wpusers.py
index 82d98d4..b3c06a2 100644
--- a/modules/discovery/users/wpusers.py
+++ b/modules/discovery/users/wpusers.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 import re
 
 class wpusers:
@@ -59,4 +59,4 @@ class wpusers:
 			if login_new == []:
 				self.printf.ipri("Not found usernames",color="r")
 		except Exception as error:
-			self.printf.ipri("Not found usernames",color="r")
\ No newline at end of file
+			self.printf.ipri("Not found usernames",color="r")
diff --git a/modules/discovery/wpall.py b/modules/discovery/wpall.py
index 2770cf3..4b5c8f5 100644
--- a/modules/discovery/wpall.py
+++ b/modules/discovery/wpall.py
@@ -19,8 +19,8 @@
 # along with WPSeku; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-from lib import wphttp
-from lib import wpprint
+from wpseku.lib import wphttp
+from wpseku.lib import wpprint
 from generic import wpgeneric 
 from plugins import wpplugin
 from themes import wptheme
@@ -31,4 +31,4 @@ class wpall:
 		wpgeneric.wpgeneric().run(agent,proxy,redirect,url)
 		wptheme.wptheme(agent,proxy,redirect,url).run()
 		wpplugin.wpplugin(agent,proxy,redirect,url).run()
-		wpusers.wpusers(agent,proxy,redirect,url).run()
\ No newline at end of file
+		wpusers.wpusers(agent,proxy,redirect,url).run()
diff --git a/setup.py b/setup.py
new file mode 100644
index 0000000..adc9ae4
--- /dev/null
+++ b/setup.py
@@ -0,0 +1,35 @@
+from distutils.core import setup
+
+setup(name='wpseku',
+      version='0.2.1',
+      author='Momo Outaadl (M4all0k)',
+      license='GPL-3',
+      description='Black box WordPress vulnerability scanner',
+      url='https://github.com/m4ll0k/WPSeku',
+      packages=[
+          'wpseku',
+          'wpseku.data',
+          'wpseku.lib',
+          'wpseku.modules',
+          'wpseku.modules.attack',
+          'wpseku.modules.bruteforce',
+          'wpseku.modules.discovery',
+          'wpseku.modules.discovery.generic',
+          'wpseku.modules.discovery.plugins',
+          'wpseku.modules.discovery.themes',
+          'wpseku.modules.discovery.users',
+      ],
+      package_dir={
+          'wpseku.data': 'data',
+          'wpseku.lib': 'lib',
+          'wpseku.modules': 'modules',
+      },
+      package_data={
+          'wpseku.data': [
+              '*.txt',
+          ],
+      },
+      scripts=[
+          'bin/wpseku',
+      ],
+)
diff --git a/wpseku/__init__.py b/wpseku/__init__.py
new file mode 100644
index 0000000..e69de29
