--- a/setup.py	2017-05-03 23:47:05.257432601 +0200
+++ b/setup.py	2017-05-03 23:47:25.688511295 +0200
@@ -51,9 +51,6 @@
         continue
     extras_require['all'].extend(reqs)
 
-setup_requires = [
-    'pytest-runner>=2.7',
-]
 
 packages = find_packages()
 
@@ -76,7 +73,6 @@
     packages=['dictdiffer'],
     zip_safe=False,
     extras_require=extras_require,
-    setup_requires=setup_requires,
     tests_require=tests_require,
     classifiers=[
         'Programming Language :: Python :: 2',
