commit bb04db3af5e533537c44fd759d151c8c8811c4bb
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Sat Jan 4 23:29:03 2020 +0100

    Fix API calls for new 'useuefi' parameter

diff --git a/oz/Debian.py b/oz/Debian.py
index 2f076fd..241effa 100644
--- a/oz/Debian.py
+++ b/oz/Debian.py
@@ -100,7 +100,8 @@ class DebianGuest(oz.Linux.LinuxCDGuest):
         if diskbus is None:
             diskbus = self.config.default_diskbus
         oz.Linux.LinuxCDGuest.__init__(self, tdl, config, auto, output_disk,
-                                       netdev, diskbus, True, True, macaddress)
+                                       netdev, diskbus, True, True, macaddress,
+                                       False)
 
         self.crond_was_active = False
         self.sshd_was_active = False
diff --git a/oz/FedoraCore.py b/oz/FedoraCore.py
index 6dce981..f6370df 100644
--- a/oz/FedoraCore.py
+++ b/oz/FedoraCore.py
@@ -62,7 +62,7 @@ class FedoraCoreGuest(oz.RedHat.RedHatLinuxCDGuest):
         oz.RedHat.RedHatLinuxCDGuest.__init__(self, tdl, config, auto,
                                               output_disk, netdev, diskbus,
                                               True, True, self.config.initrdtype,
-                                              macaddress)
+                                              macaddress, False)
 
         # FIXME: if doing an ISO install, we have to check that the ISO passed
         # in is the DVD, not the CD (since we can't change disks midway)
diff --git a/oz/Mageia.py b/oz/Mageia.py
index 1c9a518..d5dc318 100644
--- a/oz/Mageia.py
+++ b/oz/Mageia.py
@@ -90,7 +90,7 @@ class MageiaGuest(oz.Linux.LinuxCDGuest):
             diskbus = self.config.default_diskbus
         oz.Linux.LinuxCDGuest.__init__(self, tdl, config, auto, output_disk,
                                        netdev, diskbus, True, True,
-                                       macaddress)
+                                       macaddress, False)
 
         self.sshd_was_active = False
         self.crond_was_active = False
diff --git a/oz/Mandrake.py b/oz/Mandrake.py
index cae6892..0ec8a08 100644
--- a/oz/Mandrake.py
+++ b/oz/Mandrake.py
@@ -68,7 +68,8 @@ class MandrakeGuest(oz.Guest.CDGuest):
     def __init__(self, tdl, config, auto, output_disk, netdev, diskbus,
                  macaddress):
         oz.Guest.CDGuest.__init__(self, tdl, config, auto, output_disk, netdev,
-                                  None, None, diskbus, True, False, macaddress)
+                                  None, None, diskbus, True, False, macaddress,
+                                  False)
 
         if self.tdl.arch != "i386":
             raise oz.OzException.OzException("Mandrake only supports i386 architecture")
diff --git a/oz/Mandriva.py b/oz/Mandriva.py
index ab135eb..5da9b97 100644
--- a/oz/Mandriva.py
+++ b/oz/Mandriva.py
@@ -58,7 +58,8 @@ class MandrivaGuest(oz.Guest.CDGuest):
     def __init__(self, tdl, config, auto, output_disk, netdev, diskbus,
                  macaddress):
         oz.Guest.CDGuest.__init__(self, tdl, config, auto, output_disk, netdev,
-                                  None, None, diskbus, True, False, macaddress)
+                                  None, None, diskbus, True, False, macaddress,
+                                  False)
 
         self.mandriva_arch = self.tdl.arch
         if self.mandriva_arch == "i386":
diff --git a/oz/OpenSUSE.py b/oz/OpenSUSE.py
index 80d1f4e..8484abb 100644
--- a/oz/OpenSUSE.py
+++ b/oz/OpenSUSE.py
@@ -138,7 +138,7 @@ class OpenSUSEGuest(oz.Linux.LinuxCDGuest):
             diskbus = self.config.default_diskbus
         oz.Linux.LinuxCDGuest.__init__(self, tdl, config, auto, output_disk,
                                        nicmodel, diskbus, True, False,
-                                       macaddress)
+                                       macaddress, False)
 
         self.crond_was_active = False
         self.sshd_was_active = False
diff --git a/oz/RHEL_2_1.py b/oz/RHEL_2_1.py
index 66a91d5..3f61f71 100644
--- a/oz/RHEL_2_1.py
+++ b/oz/RHEL_2_1.py
@@ -32,7 +32,7 @@ class RHEL21Guest(oz.RedHat.RedHatFDGuest):
     def __init__(self, tdl, config, auto, output_disk, netdev, diskbus,
                  macaddress):
         oz.RedHat.RedHatFDGuest.__init__(self, tdl, config, auto, output_disk,
-                                         netdev, diskbus, macaddress)
+                                         netdev, diskbus, macaddress, False)
 
     def get_auto_path(self):
         """
diff --git a/oz/RHEL_3.py b/oz/RHEL_3.py
index 006a605..c8537c6 100644
--- a/oz/RHEL_3.py
+++ b/oz/RHEL_3.py
@@ -45,7 +45,7 @@ class RHEL3Guest(oz.RedHat.RedHatLinuxCDGuest):
         oz.RedHat.RedHatLinuxCDGuest.__init__(self, tdl, config, auto,
                                               output_disk, netdev, diskbus,
                                               iso_support, True, None,
-                                              macaddress)
+                                              macaddress, False)
 
         # override the sshd_config value set in RedHatLinuxCDGuest.__init__
         self.sshd_config = """\
diff --git a/oz/RHEL_4.py b/oz/RHEL_4.py
index 0bf398a..563dff8 100644
--- a/oz/RHEL_4.py
+++ b/oz/RHEL_4.py
@@ -71,7 +71,8 @@ class RHEL4Guest(oz.RedHat.RedHatLinuxCDGuest):
         # it looks ugly so for now we will just always use the boot.iso method
         oz.RedHat.RedHatLinuxCDGuest.__init__(self, tdl, config, auto,
                                               output_disk, nicmodel, diskbus,
-                                              True, True, None, macaddress)
+                                              True, True, None, macaddress,
+                                              False)
 
     def _modify_iso(self):
         """
diff --git a/oz/RHEL_5.py b/oz/RHEL_5.py
index 209a2c4..04f3ef9 100644
--- a/oz/RHEL_5.py
+++ b/oz/RHEL_5.py
@@ -80,7 +80,7 @@ class RHEL5Guest(oz.RedHat.RedHatLinuxCDYumGuest):
         oz.RedHat.RedHatLinuxCDYumGuest.__init__(self, tdl, config, auto,
                                                  output_disk, nicmodel, diskbus,
                                                  True, True, "cpio", macaddress,
-                                                 True)
+                                                 True, False)
 
     def _modify_iso(self):
         """
diff --git a/oz/RHEL_6.py b/oz/RHEL_6.py
index f6bd4c2..2b1d863 100644
--- a/oz/RHEL_6.py
+++ b/oz/RHEL_6.py
@@ -35,7 +35,7 @@ class RHEL6Guest(oz.RedHat.RedHatLinuxCDYumGuest):
         oz.RedHat.RedHatLinuxCDYumGuest.__init__(self, tdl, config, auto,
                                                  output_disk, netdev, diskbus,
                                                  True, True, "cpio", macaddress,
-                                                 True)
+                                                 True, False)
 
     def _modify_iso(self):
         """
diff --git a/oz/RHEL_7.py b/oz/RHEL_7.py
index 6e199db..512e6a9 100644
--- a/oz/RHEL_7.py
+++ b/oz/RHEL_7.py
@@ -35,7 +35,7 @@ class RHEL7Guest(oz.RedHat.RedHatLinuxCDYumGuest):
         oz.RedHat.RedHatLinuxCDYumGuest.__init__(self, tdl, config, auto,
                                                  output_disk, netdev, diskbus,
                                                  True, True, "cpio", macaddress,
-                                                 True)
+                                                 True, False)
         self.virtio_channel_name = 'org.fedoraproject.anaconda.log.0'
 
     def _modify_iso(self):
diff --git a/oz/RHL.py b/oz/RHL.py
index 0ff9aae..8e60ed3 100644
--- a/oz/RHL.py
+++ b/oz/RHL.py
@@ -38,7 +38,8 @@ class RHL9Guest(oz.RedHat.RedHatLinuxCDGuest):
         # after unpacking the initrd
         oz.RedHat.RedHatLinuxCDGuest.__init__(self, tdl, config, auto,
                                               output_disk, netdev, diskbus,
-                                              False, True, None, macaddress)
+                                              False, True, None, macaddress,
+                                              False)
 
         if self.tdl.arch != "i386":
             raise oz.OzException.OzException("Invalid arch " + self.tdl.arch + "for RHL guest")
@@ -86,7 +87,7 @@ class RHL7xand8Guest(oz.RedHat.RedHatFDGuest):
     def __init__(self, tdl, config, auto, output_disk, nicmodel, diskbus,
                  macaddress):
         oz.RedHat.RedHatFDGuest.__init__(self, tdl, config, auto, output_disk,
-                                         nicmodel, diskbus, macaddress)
+                                         nicmodel, diskbus, macaddress, False)
 
     def get_auto_path(self):
         return oz.ozutil.generate_full_auto_path("RedHatLinux" + self.tdl.update + ".auto")
diff --git a/oz/Ubuntu.py b/oz/Ubuntu.py
index 1f66826..e386809 100644
--- a/oz/Ubuntu.py
+++ b/oz/Ubuntu.py
@@ -699,7 +699,7 @@ class UbuntuGuest(oz.Linux.LinuxCDGuest):
 
         oz.Linux.LinuxCDGuest.__init__(self, tdl, config, auto, output_disk,
                                        nicmodel, diskbus, True, True,
-                                       macaddress)
+                                       macaddress, False)
 
         self.crond_was_active = False
         self.sshd_was_active = False
diff --git a/oz/Windows.py b/oz/Windows.py
index 78948ac..0651f2c 100644
--- a/oz/Windows.py
+++ b/oz/Windows.py
@@ -39,7 +39,7 @@ class Windows(oz.Guest.CDGuest):
                  macaddress):
         oz.Guest.CDGuest.__init__(self, tdl, config, auto, output_disk,
                                   netdev, "localtime", "usb", diskbus, True,
-                                  False, macaddress)
+                                  False, macaddress, False)
 
         if self.tdl.key is None:
             raise oz.OzException.OzException("A key is required when installing Windows")
