commit a723634830bcb67cd0b3ed5c958f7744cf8a39a2
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Tue Jul 18 22:45:45 2017 +0200

    Replace genisoimage with mkisofs

diff --git a/oz/Debian.py b/oz/Debian.py
index 2f076fd..2105063 100644
--- a/oz/Debian.py
+++ b/oz/Debian.py
@@ -210,7 +210,7 @@ critical initrd=%s/initrd.gz --
         Method to create a new ISO based on the modified CD/DVD.
         """
         self.log.info("Generating new ISO")
-        oz.ozutil.subprocess_check_output(["genisoimage", "-r", "-V", "Custom",
+        oz.ozutil.subprocess_check_output(["mkisofs", "-r", "-V", "Custom",
                                            "-J", "-l", "-no-emul-boot",
                                            "-b", "isolinux/isolinux.bin",
                                            "-c", "isolinux/boot.cat",
diff --git a/oz/FreeBSD.py b/oz/FreeBSD.py
index 5d799ff..88ed9c5 100644
--- a/oz/FreeBSD.py
+++ b/oz/FreeBSD.py
@@ -78,7 +78,7 @@ class FreeBSD(oz.Guest.CDGuest):
         Method to create a new ISO based on the modified CD/DVD.
         """
         self.log.debug("Generating new ISO")
-        oz.ozutil.subprocess_check_output(["genisoimage",
+        oz.ozutil.subprocess_check_output(["mkisofs",
                                            "-R", "-no-emul-boot",
                                            "-b", "boot/cdboot", "-v",
                                            "-o", self.output_iso,
diff --git a/oz/Mageia.py b/oz/Mageia.py
index 1c9a518..87628b8 100644
--- a/oz/Mageia.py
+++ b/oz/Mageia.py
@@ -256,7 +256,7 @@ label customiso
         """
         self.log.info("Generating new ISO")
 
-        oz.ozutil.subprocess_check_output(["genisoimage", "-r", "-V", "Custom",
+        oz.ozutil.subprocess_check_output(["mkisofs", "-r", "-V", "Custom",
                                            "-J", "-l", "-no-emul-boot",
                                            "-b", self.isolinuxbin,
                                            "-c", "boot.catalog",
diff --git a/oz/Mandrake.py b/oz/Mandrake.py
index cae6892..49e2d62 100644
--- a/oz/Mandrake.py
+++ b/oz/Mandrake.py
@@ -136,7 +136,7 @@ label customiso
         """
         self.log.info("Generating new ISO")
         if self.config.old_isolinux:
-            oz.ozutil.subprocess_check_output(["genisoimage", "-r", "-V", "Custom",
+            oz.ozutil.subprocess_check_output(["mkisofs", "-r", "-V", "Custom",
                                                "-J", "-cache-inodes",
                                                "-b", "Boot/cdrom.img",
                                                "-c", "Boot/boot.cat",
@@ -144,7 +144,7 @@ label customiso
                                                self.iso_contents],
                                               printfn=self.log.debug)
         else:
-            oz.ozutil.subprocess_check_output(["genisoimage", "-r", "-V", "Custom",
+            oz.ozutil.subprocess_check_output(["mkisofs", "-r", "-V", "Custom",
                                                "-J", "-l", "-no-emul-boot",
                                                "-b", "isolinux/isolinux.bin",
                                                "-c", "isolinux/boot.cat",
diff --git a/oz/Mandriva.py b/oz/Mandriva.py
index ab135eb..840b6f0 100644
--- a/oz/Mandriva.py
+++ b/oz/Mandriva.py
@@ -120,7 +120,7 @@ label customiso
         isolinuxbin = os.path.join(isolinuxdir, "isolinux/isolinux.bin")
         isolinuxboot = os.path.join(isolinuxdir, "isolinux/boot.cat")
 
-        oz.ozutil.subprocess_check_output(["genisoimage", "-r", "-V", "Custom",
+        oz.ozutil.subprocess_check_output(["mkisofs", "-r", "-V", "Custom",
                                            "-J", "-l", "-no-emul-boot",
                                            "-b", isolinuxbin,
                                            "-c", isolinuxboot,
diff --git a/oz/OpenSUSE.py b/oz/OpenSUSE.py
index 80d1f4e..e81bd4c 100644
--- a/oz/OpenSUSE.py
+++ b/oz/OpenSUSE.py
@@ -200,7 +200,7 @@ class OpenSUSEGuest(oz.Linux.LinuxCDGuest):
         Method to create a new ISO based on the modified CD/DVD.
         """
         self.log.info("Generating new ISO")
-        oz.ozutil.subprocess_check_output(["genisoimage", "-r", "-V", "Custom",
+        oz.ozutil.subprocess_check_output(["mkisofs", "-r", "-V", "Custom",
                                            "-J", "-no-emul-boot",
                                            "-b", "boot/" + self.tdl.arch + "/loader/isolinux.bin",
                                            "-c", "boot/" + self.tdl.arch + "/loader/boot.cat",
diff --git a/oz/RedHat.py b/oz/RedHat.py
index 868b549..5a7e907 100644
--- a/oz/RedHat.py
+++ b/oz/RedHat.py
@@ -93,7 +93,7 @@ Subsystem	sftp	/usr/libexec/openssh/sftp-server
         Method to create a new ISO based on the modified CD/DVD.
         """
         self.log.debug("Generating new ISO")
-        oz.ozutil.subprocess_check_output(["genisoimage", "-r", "-T", "-J", "-joliet-long",
+        oz.ozutil.subprocess_check_output(["mkisofs", "-r", "-T", "-J", "-joliet-long",
                                            "-V", "Custom", "-no-emul-boot",
                                            "-b", "isolinux/isolinux.bin",
                                            "-c", "isolinux/boot.cat",
diff --git a/oz/Ubuntu.py b/oz/Ubuntu.py
index 1f66826..265d653 100644
--- a/oz/Ubuntu.py
+++ b/oz/Ubuntu.py
@@ -844,7 +844,7 @@ PROMPT 0
         Method to create a new ISO based on the modified CD/DVD.
         """
         self.log.info("Generating new ISO")
-        oz.ozutil.subprocess_check_output(["genisoimage", "-r", "-V", "Custom",
+        oz.ozutil.subprocess_check_output(["mkisofs", "-r", "-V", "Custom",
                                            "-J", "-l", "-no-emul-boot",
                                            "-b", "isolinux/isolinux.bin",
                                            "-c", "isolinux/boot.cat",
diff --git a/oz/Windows.py b/oz/Windows.py
index 78948ac..de14f2f 100644
--- a/oz/Windows.py
+++ b/oz/Windows.py
@@ -66,7 +66,7 @@ class Windows_v5(Windows):
         Method to create a new ISO based on the modified CD/DVD.
         """
         self.log.debug("Generating new ISO")
-        oz.ozutil.subprocess_check_output(["genisoimage",
+        oz.ozutil.subprocess_check_output(["mkisofs",
                                            "-b", "cdboot/boot.bin",
                                            "-no-emul-boot", "-boot-load-seg",
                                            "1984", "-boot-load-size", "4",
@@ -167,9 +167,9 @@ class Windows_v6(Windows):
         Method to create a new ISO based on the modified CD/DVD.
         """
         self.log.debug("Generating new ISO")
-        # NOTE: Windows 2008 is very picky about which arguments to genisoimage
+        # NOTE: Windows 2008 is very picky about which arguments to mkisofs
         # will generate a bootable CD, so modify these at your own risk
-        oz.ozutil.subprocess_check_output(["genisoimage",
+        oz.ozutil.subprocess_check_output(["mkisofs",
                                            "-b", "cdboot/boot.bin",
                                            "-no-emul-boot", "-c", "BOOT.CAT",
                                            "-iso-level", "2", "-J", "-l", "-D",
@@ -252,7 +252,7 @@ class Windows_v10(Windows):
         Method to create a new ISO based on the modified CD/DVD.
         """
         self.log.debug("Generating new ISO")
-        oz.ozutil.subprocess_check_output(["genisoimage",
+        oz.ozutil.subprocess_check_output(["mkisofs",
                                            "-b", "cdboot/boot.bin",
                                            "-no-emul-boot", "-c", "BOOT.CAT",
                                            "-iso-level", "2", "-J", "-l", "-D",
