From 44e654d2d1bb7057798cab2f54b8bd01eef542b6 Mon Sep 17 00:00:00 2001
From: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date: Mon, 13 May 2019 14:55:18 +0200
Subject: [PATCH 1/2] Read ruby executable from 'ruby_exec' variable

---
 templates/sudoers.rb.erb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/templates/sudoers.rb.erb b/templates/sudoers.rb.erb
index e3e731a..92fca43 100644
--- a/templates/sudoers.rb.erb
+++ b/templates/sudoers.rb.erb
@@ -1,4 +1,4 @@
-#!<%= cmd_paths['ruby'] %>
+#!<%= ruby_exec %>
 # Automatically created by vagrant-lxc
 
 class Whitelist
-- 
2.21.0

From 09733f8fcbb8a454845768569ac2ca8beee8e740 Mon Sep 17 00:00:00 2001
From: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date: Mon, 13 May 2019 14:55:43 +0200
Subject: [PATCH 2/2] Use 'vagrant-lxc-wrapper' from scripts directory

---
 lib/vagrant-lxc/sudo_wrapper.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/vagrant-lxc/sudo_wrapper.rb b/lib/vagrant-lxc/sudo_wrapper.rb
index 910d6ff..5120edd 100644
--- a/lib/vagrant-lxc/sudo_wrapper.rb
+++ b/lib/vagrant-lxc/sudo_wrapper.rb
@@ -7,7 +7,7 @@ module Vagrant
       attr_reader :wrapper_path
 
       def self.dest_path
-        "/usr/local/bin/vagrant-lxc-wrapper"
+        Vagrant::LXC.source_root.join('scripts/vagrant-lxc-wrapper').to_s
       end
 
       def initialize(privileged: true)
-- 
2.21.0

