commit 1bc77c20621215dd7906b98ae1efdbf1e8124e5f
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Sun Jan 7 16:25:48 2018 +0100

    Fix machinectl path

diff --git a/mock/py/mockbuild/util.py b/mock/py/mockbuild/util.py
index fc484ae..ecba301 100644
--- a/mock/py/mockbuild/util.py
+++ b/mock/py/mockbuild/util.py
@@ -198,7 +198,7 @@ def get_machinectl_uuid(chroot_path):
     # we will ignore errors in machinectl, it sometimes fails for various errors (cannot find IP addr...)
     # we do not care about exit code, we just want the output
     # RHEL7 does not know --no-legend, so we must filter the legend out
-    vm_list = _safe_check_output(["/usr/bin/machinectl", "list", "--no-pager"])
+    vm_list = _safe_check_output(["/bin/machinectl", "list", "--no-pager"])
     if (isinstance(vm_list, bytes)):
         vm_list = vm_list.decode("utf-8")
     vm_list = '\n'.join(vm_list.split('\n')[1:-2])
@@ -206,7 +206,7 @@ def get_machinectl_uuid(chroot_path):
         if len(name) > 0:
             m_uuid = name.split()[0]
             try:
-                vm_root = _safe_check_output(["/usr/bin/machinectl", "show", "-pRootDirectory", m_uuid])
+                vm_root = _safe_check_output(["/bin/machinectl", "show", "-pRootDirectory", m_uuid])
                 if (isinstance(vm_root, bytes)):
                     vm_root = vm_root.decode("utf-8")
             except subprocess.CalledProcessError:
@@ -236,7 +236,7 @@ def orphansKill(rootToKill):
         m_uuid = get_machinectl_uuid(rootToKill)
         if m_uuid:
             getLog().warning("Machine %s still running. Killing...", m_uuid)
-            os.system("/usr/bin/machinectl terminate %s" % m_uuid)
+            os.system("/bin/machinectl terminate %s" % m_uuid)
 
 
 @traceLog()
commit 95560303b437fae32c2982dec8107ccbf35c2465
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Sun Jan 7 17:03:19 2018 +0100

    Fix df path

diff --git a/mock/py/mockbuild/plugins/hw_info.py b/mock/py/mockbuild/plugins/hw_info.py
index 1b17743..41f59db 100644
--- a/mock/py/mockbuild/plugins/hw_info.py
+++ b/mock/py/mockbuild/plugins/hw_info.py
@@ -47,7 +47,7 @@ class HwInfo(object):
         out.write("\n\nMemory:\n")
         out.write(output)
 
-        cmd = ["/usr/bin/df", "-H", self.buildroot.make_chroot_path()]
+        cmd = ["/bin/df", "-H", self.buildroot.make_chroot_path()]
         output = mockbuild.util.do(cmd, shell=False, returnOutput=True, raiseExc=False)
         out.write("\n\nStorage:\n")
         out.write(output)
