commit 3dbb451831e7de56283a6d22d87f4ac53d5f881a
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Sat Sep 14 18:29:11 2019 +0200

    Make spell optional

diff --git a/meson.build b/meson.build
index 4d3b57e4a..d3e092367 100644
--- a/meson.build
+++ b/meson.build
@@ -63,7 +63,8 @@ libpeas_gtk_dep = dependency('libpeas-gtk-1.0', version: libpeas_req)
 
 libgd_dep = libgd_subproject.get_variable('libgd_dep')
 
-gspell_dep = dependency('gspell-1', version: '>= 1.0', required: true)
+with_gspell = get_option('spell')
+gspell_dep = dependency('gspell-1', version: '>= 1.0', required: false)
 x11_dep = dependency('x11', required: false)
 
 introspection_dep = dependency('gobject-introspection-1.0', required: false)
@@ -154,6 +155,7 @@ summary = [
   '        User documentation:    @0@'.format(get_option('user_documentation')),
   '        GObject Introspection: @0@'.format(generate_gir),
   '        Vala API:              @0@'.format(generate_vapi),
+  '        Spell:                 @0@'.format(with_gspell),
   '',
 ]
 message('\n'.join(summary))
diff --git a/meson_options.txt b/meson_options.txt
index d0cbf0720..a3b8e82d2 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -10,6 +10,10 @@ option('gtk_doc',
        type: 'boolean', value: false,
        description: 'Build API reference for plugins (requires gtk-doc)')
 
+option('spell',
+        type:'boolean', value: true,
+        description: 'Build spell support')
+
 # This option exists for the developers, to speed up the install.
 option('user_documentation',
        type: 'boolean', value: true,
diff --git a/plugins/meson.build b/plugins/meson.build
index a9167f8d6..639539537 100644
--- a/plugins/meson.build
+++ b/plugins/meson.build
@@ -28,5 +28,7 @@ subdir('quickhighlight')
 subdir('quickopen')
 subdir('snippets')
 subdir('sort')
-subdir('spell')
+if with_gspell
+  subdir('spell')
+endif
 subdir('time')
