From: Ryan Kelly <rfkelly@mozilla.com>
Date: Tue, 2 Aug 2016 15:18:27 +1000
Subject: [PATCH] Use randomly-generated fxa.metrics_uid_secret_key for
 self-hosters.

---
 syncserver/__init__.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/syncserver/__init__.py b/syncserver/__init__.py
--- a/syncserver/__init__.py
+++ b/syncserver/__init__.py
@@ -93,6 +93,11 @@ def includeme(config):
         root_logger = logging.getLogger("")
         if not root_logger.handlers:
             logging.basicConfig(level=logging.INFO)
+    if "fxa.metrics_uid_secret_key" not in settings:
+        # Default to a randomly-generated secret.
+        # This setting isn't useful in a self-hosted setup
+        # and setting a default avoids scary-sounding warnings.
+        settings["fxa.metrics_uid_secret_key"] = os.urandom(16).encode("hex")
 
     # Include the relevant sub-packages.
     config.scan("syncserver")
