commit 625d1424cac33e022815308f3e256c2f2280f7d3
Author: Pierre ROUDIER <contact@pierreroudier.net>
Date:   Thu Jun 27 21:36:23 2013 +0200

    xz support for tar handler

diff --git a/handlers/tar.helper.in b/handlers/tar.helper.in
index 4a483be..91fec34 100644
--- a/handlers/tar.helper.in
+++ b/handlers/tar.helper.in
@@ -25,7 +25,8 @@ tar_wizard() {
        "none"     "do not filter trough" off \
        "compress" "filter trough compress" off \
        "gzip"     "filter trough gzip" off \
-       "bzip"     "filter trough bzip" on
+       "bzip"     "filter trough bzip" on \
+       "xz"       "filter trough xz" off
    [ $? = 1 ] && return;
    result="$REPLY"
    tar_compress="compress = $REPLY "
diff --git a/handlers/tar.in b/handlers/tar.in
index b4f8c58..eb5942b 100644
--- a/handlers/tar.in
+++ b/handlers/tar.in
@@ -48,6 +48,10 @@ case $compress in
       compress_option="-j"
       EXTENSION="tar.bz2"
       ;;
+   "xz")
+      compress_option="-J"
+      EXTENSION="tar.xz"
+      ;;
    "none")
       compress_option=""
       ;;
