commit e96a894bb85b72253a2a1e69e36f4f6e89713078
Author: intrigeri <intrigeri@boum.org>
Date:   Sat Feb 22 13:34:06 2014 +0000

    mysql: fix non-qualified table name extraction (Closes: Redmine#4373).

diff --git a/ChangeLog b/ChangeLog
index 91bd7a9..4d3413d 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -8,6 +8,7 @@ version 1.0.2  -- UNRELEASED
 	 . Gracefully handle legacy spaces between -o and IdentityFile.
 	mysql:
 	 . Make "nodata" option compatible with compress=no.
+	 . Fix non-qualified table name extraction. (Closes: Redmine#4373)
 	rdiff:
 	 . Add option to include rdiff-backup output in reports.
 	   Thanks to David Gasaway <dave@gasaway.org> for the patch!
diff --git a/handlers/mysql.in b/handlers/mysql.in
index 9472b8e..ceed620 100644
--- a/handlers/mysql.in
+++ b/handlers/mysql.in
@@ -276,7 +276,7 @@ then
          DUMP_STRUCT="$DUMP_BASE --no-data $db"
          for qualified_table in $nodata
          do
-            table=$( expr match "$qualified_table" "$db\.\([^\w]*\)" )
+            table=$( expr match "$qualified_table" "$db\.\(.\+\)" )
             DUMP_STRUCT="$DUMP_STRUCT $table"
          done
          DUMP="( $DUMP; $DUMP_STRUCT )"
