commit 29e2c75ede42b329d38dbf843ac35afb9148305c
Author: intrigeri <intrigeri@boum.org>
Date:   Thu Dec 26 15:42:22 2013 +0000

    mysql: make "nodata" option compatible with compress=no.

diff --git a/handlers/mysql.in b/handlers/mysql.in
index 6ade49e..9472b8e 100644
--- a/handlers/mysql.in
+++ b/handlers/mysql.in
@@ -302,7 +302,7 @@ then
          if [ "$compress" == "yes" ]; then
             execstr="$DUMP | $GZIP $GZIP_OPTS > '$dumpdir/${db}.sql.gz'"
          else
-            execstr="$DUMP -r '$dumpdir/${db}.sql'"
+            execstr="$DUMP > '$dumpdir/${db}.sql'"
          fi
       fi
       debug "su $user -c \"$execstr\""
