From 8438bb4dc863e218378f117af9720d36e36af8b4 Mon Sep 17 00:00:00 2001
From: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date: Tue, 1 Mar 2016 23:23:47 +0100
Subject: [PATCH 1/2] sys: Add support for Gentoo

---
 handlers/sys.in | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/handlers/sys.in b/handlers/sys.in
index 1d47977..ba0e94c 100644
--- a/handlers/sys.in
+++ b/handlers/sys.in
@@ -58,6 +58,11 @@ then
    os=suse
    debug "SuSE detected"
    osversion="/etc/SuSE-release"
+elif [ -f /etc/gentoo-release ]
+then
+   os=gentoo
+   debug "Gentoo detected"
+   osversion="/etc/gentoo-release"
 else
    warning "Unknown OS detected!"
 fi
@@ -91,6 +96,11 @@ then
    getconf packagesfile  $parentdir/rpmpackages.txt
    getconf packagemgr   `which rpm`
    getconf packagemgroptions   ' -qa '
+elif [ "$os" = "gentoo" ]
+then
+   getconf packagesfile  $parentdir/ebuilds.txt
+   getconf packagemgr   `which epm`
+   getconf packagemgroptions   ' -qa '
 else
    getconf packagesfile $parentdir/unknownOS.txt
 fi
-- 
2.4.10

