commit 5bc3e60a6789b4d50a9716a0508ab912886ec68a
Author: Silvio Rhatto <rhatto@riseup.net>
Date:   Mon Apr 7 17:47:22 2014 -0300

    Rsync: restoring IFS/noglob (rebase issue)

diff --git a/handlers/rsync.in b/handlers/rsync.in
index ba8a521..fcf7be8 100644
--- a/handlers/rsync.in
+++ b/handlers/rsync.in
@@ -282,6 +282,8 @@ function eval_config {
      str="${i//__star__/*}"
      excludes="${excludes} --exclude='$str'"
   done
+  IFS=$SAVEIFS
+  set +o noglob
 
   # Make sure we'll run bash at the destination
   ssh_cmd="$ssh_cmd /bin/bash"
