commit c649339ba41b4bc748598ca13c4363e25f6eba1f
Author: Silvio Rhatto <rhatto@riseup.net>
Date:   Wed Nov 5 11:18:11 2014 -0200

    Rsync: support for backupninja's test option (#8196)

diff --git a/handlers/rsync.in b/handlers/rsync.in
index d0c211c..000f3ea 100644
--- a/handlers/rsync.in
+++ b/handlers/rsync.in
@@ -920,6 +920,10 @@ function set_filelist {
 
 function set_rsync_options {
 
+  if [ "$test" ]; then
+    rsync_options="$rsync_options --dry-run"
+  fi
+
   if [ "$numericids" != "0" ]; then
     rsync_options="$rsync_options --numeric-ids"
   fi
