commit 48134276246a13ac1da72279997f1fac20a1c731
Author: Silvio Rhatto <rhatto@riseup.net>
Date:   Wed May 9 19:27:46 2012 -0300

    Update increment folder date (#3929)

diff --git a/handlers/rsync.in b/handlers/rsync.in
index 4905ace..ba8a521 100644
--- a/handlers/rsync.in
+++ b/handlers/rsync.in
@@ -456,6 +456,8 @@ function rotate_long {
     debug "daily.$max --> weekly.1"
     $nice mv $backuproot/daily.$max $backuproot/weekly.1
     mkdir -p $backuproot/metadata/weekly.1
+    # Update increment folder date and setup metadata
+    $touch $backuproot/weekly.1
     date +%c%n%s > $backuproot/metadata/weekly.1/rotated
     #if [ -f $backuproot/metadata/daily.$max/created  ]; then
     #   $nice mv $backuproot/metadata/daily.$max/created $backuproot/metadata/weekly.1/
@@ -467,6 +469,8 @@ function rotate_long {
     debug "weekly.$max --> monthly.1"
     $nice mv $backuproot/weekly.$max $backuproot/monthly.1
     mkdir -p $backuproot/metadata/monthly.1
+    # Update increment folder date and setup metadata
+    $touch $backuproot/monthly.1
     date +%c%n%s > $backuproot/metadata/monthly.1/rotated
     #if [ -f $backuproot/metadata/weekly.$max/created  ]; then
     #   $nice mv $backuproot/metadata/weekly.$max/created $backuproot/metadata/weekly.1/
@@ -579,6 +583,8 @@ function rotate_long_remote {
     echo "Debug: daily.\$max --> weekly.1"
     $nice mv $backuproot/daily.\$max $backuproot/weekly.1
     mkdir -p $backuproot/metadata/weekly.1
+    # Update increment folder date and setup metadata
+    $touch $backuproot/weekly.1
     date +%c%n%s > $backuproot/metadata/weekly.1/rotated
     #if [ -f $backuproot/metadata/daily.\$max/created  ]; then
     #   $nice mv $backuproot/metadata/daily.\$max/created $backuproot/metadata/weekly.1/
@@ -590,6 +596,8 @@ function rotate_long_remote {
     echo "Debug: weekly.\$max --> monthly.1"
     $nice mv $backuproot/weekly.\$max $backuproot/monthly.1
     mkdir -p $backuproot/metadata/monthly.1
+    # Update increment folder date and setup metadata
+    $touch $backuproot/monthly.1
     date +%c%n%s > $backuproot/metadata/monthly.1/rotated
     #if [ -f $backuproot/metadata/weekly.\$max/created  ]; then
     #   $nice mv $backuproot/metadata/weekly.\$max/created $backuproot/metadata/weekly.1/
