commit 16a55989aba73eccde4969a1e6d4e38c20c42301
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Sat Jan 18 15:08:41 2020 +0100

    Use local gettext-rs crate

diff --git a/fractal-gtk/Cargo.toml b/fractal-gtk/Cargo.toml
index 682c560..8b81665 100644
--- a/fractal-gtk/Cargo.toml
+++ b/fractal-gtk/Cargo.toml
@@ -58,8 +58,7 @@ version = "0.4.0"
 features = ["v0_0_10"]
 
 [dependencies.gettext-rs]
-git = "https://github.com/danigm/gettext-rs"
-branch = "no-gettext"
+path = "../../gettext-rs-no-gettext"
 features = ["gettext-system"]
 
 [dependencies.gtk]
