commit 7634ad9bef9fdf61ef47b29d2b947dbbbf58d10c
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Thu Feb 13 17:13:31 2020 +0100

    namedtuple: Avoid verbose argument for Python 3.7 compatibility

diff --git a/reclass/storage/yaml_git/__init__.py b/reclass/storage/yaml_git/__init__.py
index 06d839b..4c73998 100644
--- a/reclass/storage/yaml_git/__init__.py
+++ b/reclass/storage/yaml_git/__init__.py
@@ -42,7 +42,7 @@ def path_mangler(inventory_base_uri, nodes_uri, classes_uri):
     return nodes_uri, classes_uri
 
 
-GitMD = collections.namedtuple('GitMD', ['name', 'path', 'id'], verbose=False, rename=False)
+GitMD = collections.namedtuple('GitMD', ['name', 'path', 'id'], rename=False)
 
 
 class GitURI(object):
