From fbb8e2775d13047ec0338611008cbbfaa70ea6a7 Mon Sep 17 00:00:00 2001
From: Trevor Wood <Trevor.G.Wood@gmail.com>
Date: Wed, 29 Jan 2020 23:45:04 -0500
Subject: [PATCH] Fix class_mappings regression

Signed-off-by: Trevor Wood <Trevor.G.Wood@gmail.com>
---
 reclass/storage/yaml_fs/__init__.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/reclass/storage/yaml_fs/__init__.py b/reclass/storage/yaml_fs/__init__.py
index 3577b36..44c81b3 100644
--- a/reclass/storage/yaml_fs/__init__.py
+++ b/reclass/storage/yaml_fs/__init__.py
@@ -95,6 +95,7 @@ def get_node(self, name, settings):
         try:
             relpath = self._nodes[name]
             path = os.path.join(self.nodes_uri, relpath)
+            name = os.path.splitext(relpath)[0]
         except KeyError as e:
             raise reclass.errors.NodeNotFound(self.name, name, self.nodes_uri)
         entity = YamlData.from_file(path).get_entity(name, settings)
