commit 50bebe4888cb3e838c0ae552f27cece9d4ec1256
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Fri Jul 15 23:27:37 2022 +0200

    Add support for collections.abc in Python >=3.8

diff --git a/reclass/values/parser_funcs.py b/reclass/values/parser_funcs.py
index db34ceb..21952ca 100644
--- a/reclass/values/parser_funcs.py
+++ b/reclass/values/parser_funcs.py
@@ -14,6 +14,11 @@ import functools
 import pyparsing as pp
 import six
 
+try:
+    collectionsAbc = collections.abc
+except AttributeError:
+    collectionsAbc = collections
+
 tags = enum.Enum('Tags', ['STR', 'REF', 'INV'])
 
 _OBJ = 'OBJ'
@@ -49,7 +54,7 @@ def _asList(x):
     return x
 
 def listify(w, modifier=_asList):
-    if (isinstance(w, collections.Iterable) and
+    if (isinstance(w, collectionsAbc.Iterable) and
             not isinstance(w, six.string_types)):
         cls = type(w)
         if cls == pp.ParseResults:
