commit e477e7c0e0f5dde1c82c335da280a4e31fca7cd0
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Thu Feb 13 17:46:43 2020 +0100

    setup.py: Avoid enum34 dependency with Python >=3.5

diff --git a/requirements.txt b/requirements.txt
index 5f6aed1..a470ea7 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,5 +1,4 @@
 pyparsing
 pyyaml
 six
-enum34
 ddt
diff --git a/setup.py b/setup.py
index ab23207..aedc44a 100644
--- a/setup.py
+++ b/setup.py
@@ -42,7 +42,7 @@ setup(
     url = URL,
     packages = find_packages(exclude=['*tests']), #FIXME validate this
     entry_points = { 'console_scripts': console_scripts },
-    install_requires = ['pyparsing', 'pyyaml', 'six', 'enum34', 'ddt'], #FIXME pygit2 (require libffi-dev, libgit2-dev 0.26.x )
+    install_requires = ['pyparsing', 'pyyaml', 'six', 'ddt'], #FIXME pygit2 (require libffi-dev, libgit2-dev 0.26.x )
 
     classifiers=[
         'Development Status :: 4 - Beta',
