commit 277b968e108f00ef12444e399dd9d5fc2769f117
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Sat Nov 9 17:22:03 2019 +0100

    Add flags for disabling optional dependencies

diff --git a/deja-dup/nautilus/meson.build b/deja-dup/nautilus/meson.build
index 04b136f3..3276d675 100644
--- a/deja-dup/nautilus/meson.build
+++ b/deja-dup/nautilus/meson.build
@@ -16,7 +16,7 @@
 # You should have received a copy of the GNU General Public License
 # along with Déjà Dup.  If not, see <http://www.gnu.org/licenses/>.
 
-if nautilus_dep.found()
+if nautilus_dep.found() and get_option('enable_nautilus')
 
 # Done separately because nautilus-extension's .gir file is named differently
 # than its .pc file, and meson doesn't handle that gracefully.
diff --git a/meson.build b/meson.build
index e43d80ad..41c3238b 100644
--- a/meson.build
+++ b/meson.build
@@ -106,11 +106,11 @@ if gdk_x11_dep.found()
 common_vflags += ['--define=HAS_X11']
 endif
 
-if goa_dep.found()
+if goa_dep.found() and get_option('enable_goa')
 common_vflags += ['--define=HAS_GOA']
 endif
 
-if packagekit_dep.found()
+if packagekit_dep.found() and get_option('enable_packagekit')
 common_vflags += ['--define=HAS_PACKAGEKIT']
 endif
 
diff --git a/meson_options.txt b/meson_options.txt
index b6ce9d4c..3851f229 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -7,3 +7,7 @@ option('swiftclient_pkgs', type: 'string', value: '', description: 'Distro packa
 option('profile', type: 'string', value: '', description: 'Profile name, for parallel installations')
 
 option('google_client_id', type: 'string', value: '916137916439-evn6skqan91m96fmsskk8102e3iepv22.apps.googleusercontent.com', description: 'Google API client ID')
+
+option('enable_goa', type: 'boolean', value: true, description: 'Enable support for Gnome Online Accounts')
+option('enable_nautilus', type: 'boolean', value: true, description: 'Enable nautilus integration')
+option('enable_packagekit', type: 'boolean', value: true, description: 'Enable installation of missing packages through PackageKit')
