diff --git a/setup.py b/setup.py
index 9004c28..7dd8091 100644
--- a/setup.py
+++ b/setup.py
@@ -46,7 +46,6 @@ setup(
     license="Apache License Version 2.0",
     url="https://github.com/openshift/openshift-restclient-python",
     keywords=["Swagger", "OpenAPI", "Kubernetes", "OpenShift"],
-    install_requires=extract_requirements('requirements.txt'),
     packages=find_packages(include='openshift.*'),
     include_package_data=True,
     data_files=[
