--- a/setup.py	2017-05-05 21:03:44.767263968 +0200
+++ b/setup.py	2017-05-05 21:04:07.498441859 +0200
@@ -46,7 +46,6 @@
     license="Apache License Version 2.0",
     url="https://github.com/openshift/openshift-restclient-python",
     keywords=["Swagger", "OpenAPI", "Kubernetes", "OpenShift"],
-    install_requires=extract_requirements('requirements.txt'),
     packages=find_packages(include='openshift.*'),
     include_package_data=True,
     long_description='Python client for OpenShift http://openshift.redhat.com/',
