commit c9fe67014d2e4218497b528163b3d40d42bd883d
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Sat Mar 16 21:32:45 2019 +0100

    Allow full control over which plugins are built

diff --git a/meson_options.txt b/meson_options.txt
index d6f8b4290..ea87784ae 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,4 +1,25 @@
 option('enable-easy-codec-installation', type: 'combo', choices: ['yes', 'no', 'auto'], value: 'auto', description: 'Whether to enable easy codec installation support for GStreamer')
 option('enable-python', type: 'combo', choices: ['yes', 'no', 'auto'], value: 'auto', description: 'Enable python support')
-option('with-plugins', type: 'combo', choices: ['all', 'none', 'auto'], value: 'auto', description: 'Which Totem plugins to compile (default: auto; "all", "none" and "auto" are valid)')
+option('with-plugins', type: 'array', choices: [
+  'all',
+  'auto',
+  'apple-trailers',
+  'autoload-subtitles',
+  'dbusservice',
+  'im-status',
+  'lirc',
+  'media-player-keys',
+  'opensubtitles',
+  'properties',
+  'pythonconsole',
+  'recent',
+  'rotation',
+  'samplepython',
+  'save-file',
+  'screensaver',
+  'screenshot',
+  'skipto',
+  'variable-rate',
+  'vimeo'
+], value: ['auto'], description: 'Which Totem plugins to compile (default: auto; "all", "none" and "auto" are valid)')
 option('enable-gtk-doc', type: 'boolean', value: false, description: 'use gtk-doc to build documentation')
diff --git a/src/plugins/meson.build b/src/plugins/meson.build
index 22bfc8f4a..5a2e003a7 100644
--- a/src/plugins/meson.build
+++ b/src/plugins/meson.build
@@ -37,7 +37,7 @@ plugins = []
 
 plugins_option = get_option('with-plugins')
 if plugins_option != 'none'
-  plugins += [
+  foreach plugin: [
     'apple-trailers',
     'autoload-subtitles',
     'im-status',
@@ -53,16 +53,38 @@ if plugins_option != 'none'
     'variable-rate',
     'vimeo'
   ]
+    if plugins_option.contains('all') or plugins_option.contains('auto') or plugins_option.contains(plugin)
+      plugins += plugin
+    endif
+  endforeach
 
-  plugin_error = (plugins_option == 'all')
+  if plugins_option.contains('all') or plugins_option.contains('auto') or plugins_option.contains('lirc')
+    lirc_dep = dependency('lirc', required: false)
+  else
+    lirc_dep = disabler()
+  endif
+
+  if lirc_dep.found() and cc.has_function('lirc_init', dependencies: lirc_dep) and cc.has_header('lirc/lirc_client.h')
+    plugins += 'lirc'
+  else
+    str = 'you need lirc_client installed for the lirc plugin'
+    if plugins_option.contains('all') or plugins_option.contains('lirc')
+      error(str)
+    endif
+    message(str)
+  endif
 
   if have_python
-    plugins += [
+    foreach plugin: [
       'dbusservice',
       'opensubtitles',
       'pythonconsole',
       'samplepython'
     ]
+      if plugins_option.contains('all') or plugins_option.contains('auto') or plugins_option.contains(plugin)
+        plugins += plugin
+      endif
+    endforeach
   endif
 endif
 
