From 1cb4c984f1cb53241029400e59759f49648957bf Mon Sep 17 00:00:00 2001
From: Markus Siemens <markus@m-siemens.de>
Date: Thu, 3 Sep 2020 09:37:35 +0200
Subject: [PATCH] Tests: Make tests work with nosetest again

---
 PyGitUp/tests/__init__.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/PyGitUp/tests/__init__.py b/PyGitUp/tests/__init__.py
index 70a31f1..8b53728 100644
--- a/PyGitUp/tests/__init__.py
+++ b/PyGitUp/tests/__init__.py
@@ -3,6 +3,7 @@
 from tempfile import mkdtemp
 
 import contextlib
+import os
 from git import *
 
 basepath = mkdtemp(prefix='PyGitUp.')
