From 56286de885cfa5721a5d74b873df49238f324029 Mon Sep 17 00:00:00 2001
From: Pavla Kratochvilova <pkratoch@redhat.com>
Date: Wed, 13 Feb 2019 12:39:30 +0100
Subject: [PATCH] Revert: 9309e92332241ff1113433057c969cebf127734e

---
 libdnf/conf/ConfigMain.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libdnf/conf/ConfigMain.cpp b/libdnf/conf/ConfigMain.cpp
index fc4ff01..50c640c 100644
--- a/libdnf/conf/ConfigMain.cpp
+++ b/libdnf/conf/ConfigMain.cpp
@@ -210,7 +210,7 @@ class ConfigMain::Impl {
     OptionSeconds metadata_timer_sync{60 * 60 * 3}; // 3 hours
     OptionStringList disable_excludes{std::vector<std::string>{}};
     OptionEnum<std::string> multilib_policy{"best", {"best", "all"}}; // :api
-    OptionBool best{true}; // :api
+    OptionBool best{false}; // :api
     OptionBool install_weak_deps{true};
     OptionString bugtracker_url{BUGTRACKER};
     OptionBool zchunk{true};
--
libgit2 0.27.7

