commit cecc3c46fb6acdcf0f3496284903144831232fee
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Sat Mar 27 16:01:03 2021 +0100

    argyll: Adjust spotread/colprof path for Gentoo

diff --git a/contrib/session-helper/cd-main.c b/contrib/session-helper/cd-main.c
index 10d94a7..4ae922d 100644
--- a/contrib/session-helper/cd-main.c
+++ b/contrib/session-helper/cd-main.c
@@ -1040,7 +1040,7 @@ cd_main_generate_profile (CdMainPrivate *priv, GError **error)
 	g_autoptr(GPtrArray) array = NULL;
 
 	/* get correct name of the command */
-	command = cd_main_find_argyll_tool ("colprof", error);
+	command = cd_main_find_argyll_tool ("argyll-colprof", error);
 	if (command == NULL)
 		return FALSE;
 
diff --git a/meson.build b/meson.build
index 916eadb..07e48e6 100644
--- a/meson.build
+++ b/meson.build
@@ -160,15 +160,15 @@ if get_option('sane')
 endif
 
 if get_option('argyllcms_sensor')
-  spotread = find_program('spotread', required : false)
+  spotread = find_program('argyll-spotread', required : false)
   if spotread.found() == false
-    warning('spotread not found. Argyll sensor will not work unless argyll is installed')
+    warning('argyll-spotread not found. Argyll sensor will not work unless argyll is installed')
   endif
   conf.set('HAVE_ARGYLLCMS_SENSOR', '1')
 endif
 
 if get_option('print_profiles')
-  colprof = find_program('colprof')
+  colprof = find_program('argyll-colprof')
   conf.set_quoted('TOOL_COLPROF', colprof.path())
 endif
 
diff --git a/src/sensors/argyll/cd-sensor-argyll.c b/src/sensors/argyll/cd-sensor-argyll.c
index 21b354b..42713d1 100644
--- a/src/sensors/argyll/cd-sensor-argyll.c
+++ b/src/sensors/argyll/cd-sensor-argyll.c
@@ -251,7 +251,7 @@ cd_sensor_get_sample_async (CdSensor *sensor,
 	/* if spotread is not already running then execute */
 	if (!cd_spawn_is_running (priv->spawn)) {
 		argv = g_ptr_array_new_with_free_func (g_free);
-		g_ptr_array_add (argv, g_strdup ("spotread"));
+		g_ptr_array_add (argv, g_strdup ("argyll-spotread"));
 		g_ptr_array_add (argv, g_strdup ("-d"));
 		g_ptr_array_add (argv, g_strdup_printf ("-c%u", priv->communication_port));
 		g_ptr_array_add (argv, g_strdup ("-N")); //no autocal
