diff --git a/bin/rinse b/bin/rinse
index 51d0618..d99bc29 100755
--- a/bin/rinse
+++ b/bin/rinse
@@ -31,7 +31,7 @@ rinse - RPM Installation Entity.
    --before-post-install Additionally run the specified script before
                          the post install script.
    --post-install        Run the given post-install script instead of the
-                         default files in /usr/lib/rinse/$distro
+                         default files in /usr/libexec/rinse/$distro
 
   Misc Options:
 
@@ -46,7 +46,7 @@ rinse - RPM Installation Entity.
                          (Default is to read it from /etc/rinse/rinse.conf)
    --list-distributions  Show installable distributions.
    --print-uris          Only show the RPMs which should be downloaded.
-                         default files in /usr/lib/rinse/$distro
+                         default files in /usr/libexec/rinse/$distro
    --verbose             Enable verbose output.
 
 =cut
@@ -533,7 +533,7 @@ sub listDistributions {
         #  Get the name - so that we can test for the directory.
         #
       if ( $file =~ /^(.*)\/(.*)\.packages$/ ) {
-        push @avail, $2 if ( -d "/usr/lib/rinse/$2" );
+        push @avail, $2 if ( -d "/usr/libexec/rinse/$2" );
       }
     }
 
@@ -1234,8 +1234,8 @@ sub postInstallationCustomization {
     # common files.
     #
     my @scripts;
-    push( @scripts, "/usr/lib/rinse/common" );
-    push( @scripts, "/usr/lib/rinse/" . lc ( $distribution ) );
+    push( @scripts, "/usr/libexec/rinse/common" );
+    push( @scripts, "/usr/libexec/rinse/" . lc ( $distribution ) );
 
     #
     #  For each one
